module/VuFind/src/VuFind/AjaxHandler/GetRecordVersions.php

```diff
@@ -0,0 +1,107 @@
+ *
+ * PHP version 7
+ *
+ * Copyright (C) The National Library of Finland 2019.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  AJAX
+ * @author   Ere Maijala <ere.maijala@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+namespace VuFind\AjaxHandler;
+
+use Laminas\Mvc\Controller\Plugin\Params;
+use VuFind\Record\Loader;
+use VuFind\RecordTab\TabManager;
+use VuFind\Session\Settings as SessionSettings;
+use VuFind\View\Helper\Root\Record;
+
+/**
+ * AJAX handler for fetching versions link
+ *
+ * @category VuFind
+ * @package  AJAX
+ * @author   Ere Maijala <ere.maijala@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+class GetRecordVersions extends \VuFind\AjaxHandler\AbstractBase
+{
+    /**
+     * Record loader
+     *
+     * @var Loader
+     */
+    protected $recordLoader;
+
+    /**
+     * Record plugin
+     *
+     * @var Record
+     */
+    protected $recordPlugin;
+
+    /**
+     * Tab manager
+     *
+     * @var TabManager
+     */
+    protected $tabManager;
+
+    /**
+     * Constructor
+     *
+     * @param SessionSettings $ss     Session settings
+     * @param Loader          $loader Record loader
+     * @param Record          $rp     Record plugin
+     * @param TabManager      $tm     Tab manager
+     */
+    public function __construct(SessionSettings $ss, Loader $loader, Record $rp,
+        TabManager $tm
+    ) {
+        $this->sessionSettings = $ss;
+        $this->recordLoader = $loader;
+        $this->recordPlugin = $rp;
+        $this->tabManager = $tm;
+    }
+
+    /**
+     * Handle a request.
+     *
+     * @param Params $params Parameter helper from controller
+     *
+     * @return array [response data, HTTP status code]
+     */
+    public function handleRequest(Params $params)
+    {
+        $this->disableSessionWrites(); // avoid session write timing bug
+
+        $id = $params->fromPost('id', $params->fromQuery('id'));
+        $driver = $this->recordLoader->load($id);
+        $tabs = $this->tabManager->getTabsForRecord($driver);
+        $full = true;
+
+        $html = ($this->recordPlugin)($driver)->renderTemplate(
```

Comment:
I think in other contexts, we accomplish this sort of thing by explicitly calling `$this->recordPlugin->__invoke()` rather than using `($this->recordPlugin)()`. I mention this not because you should change this, but just because it took me a moment to wrap my head around what was going on here. This way is probably better, though. Maybe someday the other __invoke() calls should be revisited.

Reply:
I've captured this issue in [VUFIND-1456](https://vufind.org/jira/browse/VUFIND-1456) so I can resolve the thread here, since this is not really relevant to the PR in progress.